internal/services/containers/container_connected_registry_resource.go

```diff
@@ -0,0 +1,424 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type ContainerConnectedRegistryResource struct{}
+
+var _ sdk.ResourceWithUpdate = ContainerConnectedRegistryResource{}
+
+type ContainerConnectedRegistryModel struct {
+	Name                string   `tfschema:"name"`
+	ContainerRegistryId string   `tfschema:"container_registry_id"`
+	ParentRegistryId    string   `tfschema:"parent_registry_id"`
+	SyncTokenId         string   `tfschema:"sync_token_id"`
+	SyncSchedule        string   `tfschema:"sync_schedule"`
+	SyncMessageTTL      string   `tfschema:"sync_message_ttl"`
+	SyncWindow          string   `tfschema:"sync_window"`
+	Mode                string   `tfschema:"mode"`
+	NotificationList    []string `tfschema:"notification_list"`
+	ClientTokenIds      []string `tfschema:"client_token_ids"`
+	LogLevel            string   `tfschema:"log_level"`
+	AuditLogEnabled     bool     `tfschema:"audit_log_enabled"`
+}
+
+func (r ContainerConnectedRegistryResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ContainerRegistryName,
+		},
+
+		"container_registry_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.RegistryID,
+		},
+
+		"parent_registry_id": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.Any(validate.ContainerConnectedRegistryID, validate.RegistryID),
+		},
+
+		"sync_token_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ContainerRegistryTokenID,
+		},
+
+		"sync_schedule": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Default:      "* * * * *",
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"sync_message_ttl": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Default:      "P1D",
+			ValidateFunc: tfvalidate.ISO8601DurationBetween("P1D", "P90D"),
+		},
+
+		"sync_window": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: tfvalidate.ISO8601DurationBetween("PT3H", "P7D"),
+		},
+
+		"mode": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ForceNew: true,
+			Default:  string(containerregistry.ConnectedRegistryModeReadWrite),
+			ValidateFunc: validation.StringInSlice(
+				[]string{
+					string(containerregistry.ConnectedRegistryModeMirror),
+					string(containerregistry.ConnectedRegistryModeReadOnly),
+					string(containerregistry.ConnectedRegistryModeReadWrite),
+					string(containerregistry.ConnectedRegistryModeRegistry),
+				},
+				false,
+			),
+		},
+
+		"notification_list": {
```

Comment:
list is redundant, also what type of destination is expected here?
```suggestion
		"notification": {
```

also crazy idea, would it make sense to break this out into blocks validating each component? open to input on that

Reply:
@katbyte Good idea, I'll make the change.